src/transformStateWithClones.js

```diff
@@ -8,6 +8,44 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const transformedClones = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    const clone = { ...currentState };
+
+    switch (action.type) {
+      case 'addProperties':
+        for (const data in action.extraData) {
+          clone[data] = action.extraData[data];
+        }
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          for (const data in clone) {
+            if (data === key) {
+              delete clone[data];
+            }
+          }
+        }
```

Comment:
you can use only `for in` here

Reply:
why i can't use `for of`? `keysToRemove` is an array. shouldn't i use `for of` for array iteration?
@Andriy-Shymkiv